server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1306,7 +1306,9 @@ private void finalizeSnapshotEntry(SnapshotsInProgress.Entry entry, Metadata met
                                 entry.partial() ? shardGenerations.totalShards() : entry.shards().size(),
                                 shardFailures,
                                 entry.includeGlobalState(),
-                                entry.userMetadata(),
+                                // TODO: remove this hack making the metadata mutable once
+                                //       https://github.com/elastic/elasticsearch/pull/72776 has been merged
+                                entry.userMetadata() == null ? null : new HashMap<>(entry.userMetadata()),
```

Comment:
The problem with the linked test failure was that the map in the entry was mutated while a snapshot status request was iterating over it. We must not have mutable objects in the cluster state so even though this is a horrible hack it at least gives us safe behavior in how our state machines work.